bentoml/deployment/aws_lambda/__init__.py

```diff
@@ -0,0 +1,376 @@
+            )
+        except BentoMLException as error:
+            return ApplyDeploymentResponse(status=exception_to_return_status(error))
+
+    def _apply(self, deployment_pb, bento_pb, yatai_service, bento_path):
+        if loader._is_remote_path(bento_path):
+            with loader._resolve_remote_bundle_path(bento_path) as local_path:
+                return self._apply(deployment_pb, bento_pb, yatai_service, local_path)
+
+        try:
+            deployment_spec = deployment_pb.spec
+            aws_config = deployment_spec.aws_lambda_operator_config
+            bento_service_metadata = bento_pb.bento.bento_service_metadata
+
+            # TODO
+            python_runtime = 'python3.7'
+            bento_service_py_version = version.parse(
+                bento_service_metadata.env.python_version
+            )
+            if bento_service_py_version < version.parse('3.0.0'):
+                raise BentoMLException(
+                    'Python 2 is not supported for Lambda Deployment'
+                )
+            elif (
+                version.parse('3.6.0')
+                < bento_service_py_version
+                < version.parse('3.7.0')
+            ):
+                python_runtime = 'python3.6'
+
+            api_names = (
+                [aws_config.api_name]
+                if aws_config.api_name
+                else [api.name for api in bento_service_metadata.apis]
+            )
+            ensure_deploy_api_name_exists_in_bento(
+                [api.name for api in bento_service_metadata.apis], api_names
+            )
+            try:
+                is_s3_url(aws_config.s3_path)
+            except Exception:
+                raise BentoMLException(
+                    '{} is not recognizable s3 path'.format(aws_config.s3_path)
+                )
+
+            parsed_url = urlparse(aws_config.s3_path)
+            lambda_s3_bucket = parsed_url.netloc
+            deployment_path_prefix = os.path.join(
+                parsed_url.path[1:],
+                'bentoml',
+                'deployments',
+                deployment_pb.namespace,
+                deployment_pb.name,
+            )
+
+            logger.debug('Check s3 path is available or not')
+            check_s3_bucket_exists_or_raise(lambda_s3_bucket, aws_config.s3_region)
+            logger.info('Uploading artifacts to S3 bucket')
+            artifacts_prefix = os.path.join(
+                deployment_path_prefix,
+                'artifacts',
+                deployment_spec.bento_name,
+                deployment_spec.bento_version,
+            )
+            upload_bento_service_artifacts_to_s3(
+                aws_config.region,
+                lambda_s3_bucket,
+                artifacts_prefix,
+                bento_path,
+                deployment_spec.bento_name,
+            )
+            with TempDirectory() as lambda_project_dir:
+                logger.debug('Generating template.yaml for lambda project')
```

Comment:
nit: `logger.debug(Generating cloudformation template for lambda project: %s", template_file_path)`